specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,266 @@
+
+Roles to be used with amps should be suffixed with `-internal` to indicate that the role is used for amps and NOT intended to be assigned to users.
+
+## Data Hub Roles and Privileges
+
+### data-hub-common
+
+Data Hub role containing the common privileges across Data Hub roles.
+
+#### Inherited Roles
+ - data-hub-module-reader
+ - rest-reader
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ xdmp-eval-in | Required to specify database for REST Client. (Note: This privilege alone doesn't allow eval to be called. The REST API code does the neccesary amping for eval execution)
+ xdbc-eval-in | Same as xdmp-eval-in
+
+### data-hub-common-writer
+
+Data Hub role containing the common privileges across Data Hub writer roles.
+
+#### Inherited Roles
+ - data-hub-common
+ - rest-writer
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ ps-user | Needed to record provenance data
+ xdmp-document-load | Needed to load data directly in the database
+ any-uri | Required to write documents to any URI without explicit privileges granted
+ any-collection | Required to write documents to any collection without explicit privileges granted
+
+### hub-central-user
+
+All Data Hub roles intended for Hub Central users inherit this role.
+
+#### Associate Authorities
+ - canLoginToHubCentral
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-downloader
+
+Role to determine if a user can download the Hub Central project.
+
+#### Associate Authorities
+ - canDownloadProject
+
+#### Inherited Roles
+ - data-hub-common
+
+### hub-central-clear-data
+
+Role to determine if a user can clear data through Hub Central.
+
+#### Associate Authorities
+ - canClearData
+
+#### Inherited Roles
+ - data-hub-common-writer
+
+### hub-central-step-runner
+
+Can run steps through Hub Central
+
+#### Associate Authorities
+ - canReadStep
+ - canRunStep
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-common-writer
+ - data-hub-flow-reader
+ - data-hub-load-data-reader
+ - data-hub-mapping-reader
+ - data-hub-match-merge-reader
+ - data-hub-step-definition-reader
+
+### hub-central-step-writer
+
+Can create and run steps though Hub Central
+
+#### Associate Authorities
+ - canWriteStep
+
+#### Inherited Roles
+ - hub-central-step-runner
+ - data-hub-flow-writer
+
+### hub-central-load-reader
+
+Can read load data artifacts through Hub Central
+
+#### Associate Authorities
+ - canReadLoadData
+
+#### Inherited Roles
+ - hub-central-user
+ - data-hub-load-data-reader
+
+### hub-central-load-writer
+
+Can read and write load data artifacts through Hub Central
+
+#### Associate Authorities
+ - canWriteLoadData
+
+#### Inherited Roles
+ - hub-central-load-reader
+ - data-hub-common-writer
+ - data-hub-load-data-reader
```

Comment:
hub-central-load-reader inherits data-hub-load-data-reader already.

Reply:
good catch. that should be `data-hub-load-data-writer`